internal/config/ca_config/ca_config.go

```diff
@@ -0,0 +1,50 @@
+package ca_config
+
+type CAIdType int
+
+const (
+	InternalCA CAIdType = iota + 1
+	AsyncInternalCA
+)
+
+type InternalCACfg struct {
+	CaCertFile         string `json:"caCertFile,omitempty"`
+	CaKeyFile          string `json:"caKeyFile,omitempty"`
+	SignerCertName     string `json:"SignerCertName,omitempty"`
+	CaSerialFile       string `json:"caSerialFile,omitempty"`
+	CaCertValidityDays int    `json:"CaCertValidityDays,omitempty"`
+	CACertStore        string `json:"CACertStore,omitempty"`
+}
+
+type CAConfigType struct {
+	CAType                          CAIdType       `json:"CAType,omitempty"`
```

Comment:
```suggestion
	CAType                          CAIdType       `json:"type,omitempty"`
```
I'd leave out "CA", as all of this config is about CA.